### PR TITLE
Bug 1162918 - [Stingray][Home] Animation of moving card down to folde…

### PR DESCRIPTION
**Pull request by begeeben**
…r list is buggy when cardManager is longer than screen
- Prevent deck from moving to a folder.
- Fix a card moves obliquely down to the folder list bug.

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d04ed05576cfcd581858e2d64cf9baf8bc93f87f

File: shared/js/smart-screen/scrollable.js [hunk 1]
```diff
@@ -462,6 +464,10 @@
       this._setNodePosition(newIdx);
 
       this.spatialNavigator.add(itemElem);
+      if (this.listElem.classList.contains('reversed')) {
```
Comment (rexboy7) [inline shared/js/smart-screen/scrollable.js]: use `this._isReversed`

Comment (rexboy7) [inline shared/js/smart-screen/scrollable.js]: Put a "XXX" before if clause and describe why we have to do realign and shift for just `reversed` case

File: shared/js/smart-screen/scrollable.js [hunk 2]
```diff
@@ -144,11 +144,13 @@
       var distance = Math.abs(this._getScrollOffset(newItem) -
                               this.translateX);
       var duration = (distance < 960) ? 0.3 : distance / 2000;
-
-      this.listElem.style.transitionDuration = duration + 's';
+      if (!this.listElem.classList.contains('reversed')) {
```
Comment (rexboy7) [inline shared/js/smart-screen/scrollable.js]: What's the difference if we don't add this condition?

File: shared/js/smart-screen/scrollable.js [hunk 3]
```diff
@@ -144,11 +144,13 @@
...
+        this.listElem.style.transitionDuration = duration + 's';
+      }
       this.scrollTo(newItem);
       if (!prevTransform ||
           prevTransform === this.listElem.style.transform ||
+          this.listElem.classList.contains('reversed') ||
```
Comment (rexboy7) [inline shared/js/smart-screen/scrollable.js]: Do we really need this line?

File: shared/js/smart-screen/scrollable.js [hunk 4]
```diff
@@ -537,6 +543,30 @@
         'translateX(calc((100% + ' + this.spacing + 'rem) * ' + tabstop + '))';
     },
 
+    /**
+     * Shift node positions without animations.
+     * @param  {[type]} skipIdx The index of the node that needs not be shifted.
+     * @param  {[type]} offset  The unit offset. Use positive integer to shift
+     *                          right and negative integer to shift left.
+     */
+    _shiftOtherNodesPosition: function(skipIdx, offset) {
```
Comment (rexboy7) [inline shared/js/smart-screen/scrollable.js]: use `_shiftNodesPosition: function (offset, skipIdx)`

File: shared/js/smart-screen/scrollable.js [hunk 5]
```diff
@@ -537,6 +543,30 @@
...
+        if (idx !== skipIdx) {
+          node.style.transitionProperty = 'none';
+          var tabstop = this._getTabstop(idx) + offset;
+          this.getNodeFromItem(node).style.transform =
+            'translateX(calc((100% + ' +
+            this.spacing + 'rem) * ' + tabstop + '))';
+        }
+      }.bind(this));
+
+      getComputedStyle(this.nodes[0]).width;
```
Comment (rexboy7) [inline shared/js/smart-screen/scrollable.js]: Let's wrap this `getComputedStyle` as a function like `forceReflow(elem)`.
And you can replace the `getComputedStyle` at the other place by `forceReflow`.

File: shared/js/smart-screen/scrollable.js (hunk 3 above)
Comment (rexboy7) [inline shared/js/smart-screen/scrollable.js]: Let's try to use `this.refElem` instead.

File: shared/js/smart-screen/scrollable.js (hunk 2 above)
Comment (rexboy7) [inline shared/js/smart-screen/scrollable.js]: Let's remove this clause since it would be removed right away in `endSlide()` in our case.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=16a8ff43cd02d7f36859af8cf58f36b231cdfe18

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=38ac1aac49db8a204bbf314408b4dddbf9548bb3

**Pull request merged**